[Travis Toh] iP

Trav1sT opened the pull request.

Commit:
Add Gradle support

Trav1sT committed:
Level-1

Trav1sT committed:
Remove ignored files

Trav1sT committed:
Level-2

Trav1sT committed:
Add response for boundary case(empty list)

Trav1sT committed:
Add enum

Trav1sT committed:
Level-3

Trav1sT committed:
Level-4

Trav1sT committed:
Add a starting sentence to list command

Trav1sT committed:
A-TextUiTesting

Trav1sT committed:
Level5-Handle exceptions

Trav1sT committed:
Level6-Add delete cmd

Trav1sT committed:
Refactor code and add personality to the response

Trav1sT committed:
Level 7 - Save

Trav1sT committed:
Level 8 - Add LocalDateTime to Event and Deadline

Trav1sT committed:
Update to save whenever task list changes
Previously, changes are saved on exit

Trav1sT committed:
Merge branch 'branch-Level-7' into master

Trav1sT committed:
Merge and resolve conflicts

Trav1sT committed:
Add Storage, TaskList, Parser and Ui class
UI: deals with interaction with user
Storage: deals with loading tasks from the file adn saving tasks in the file
Parser: deals with making sense of the user command
TaskList: contains the task list

Trav1sT committed:
Remove ignored files

Trav1sT committed:
Package classes into bot package

Trav1sT committed:
Add Junit test for Deadline and Todo class

Trav1sT committed:
Add Javadocs

Trav1sT committed:
Amend indentations and add line breaks to comply with given std

Trav1sT committed:
Merge branch 'branch-A-JavaDoc' into master

Trav1sT committed:
Merge branch 'branch-A-CodingStandard' into master

Trav1sT committed:
Add Find command

Trav1sT committed:
Merge branch 'branch-Level-9' into master

Trav1sT committed:
Add JUnit test for Main and bug fixes
Fixed the bug where the deserialisation of task class is off by last char.
Fixed the bug where the time format is not serialised properly.

Trav1sT committed:
Refactor code
Repackage classes
Add new classes for the command

Trav1sT committed:
Code cleanup

Trav1sT committed:
Add Javadocs

Trav1sT committed:
Merge remote-tracking branch 'origin/add-gradle-support' into master

Trav1sT committed:
Configure for gradle and checkstyle

Trav1sT committed:
Start tracking config/checkstyle.xml

tanweijie123 commented on src/main/java/bot/Parser.java:
```diff
@@ -0,0 +1,176 @@
+public class Parser {
+
+    /**
+     * Returns the command based on the first character of user's input.
+     * If the command given is invalid, InvalidInputException is thrown.
+     * @param input user's input
```
Consider putting an empty line between the description and parameter section

tanweijie123 commented on src/main/java/bot/command/DeleteCommand.java:
```diff
@@ -0,0 +1,44 @@
+/**
+ * A type of command that removes Task from TaskList.
+ */
+public class DeleteCommand extends Command{
```
Missing space after Command

tanweijie123 commented on src/main/java/bot/command/ListCommand.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    /**
+     * Displayed all tasks in tasklist
```
Maybe it would sound better if it starts with "Displays"

tanweijie123 commented on src/main/java/bot/Parser.java:
```diff
@@ -0,0 +1,176 @@
+package bot;
+
+import bot.command.*;
```
Try to avoid using .* import

tanweijie123 reviewed (commented):
Looks good 👍 
A few minor coding standards to fix and you are good to go.

Trav1sT committed:
Add basic GUI

Trav1sT committed:
Merge branch 'branch-Level-10' into master

cheoksuanne commented on src/main/java/bot/Parser.java:
```diff
@@ -0,0 +1,176 @@
+     * @param input user's input
+     * @return an array of 2 String, first String is the name of deadline
+     * while second is the deadline's date
+     * @throws InvalidInputException command's argument is invalid
```
I think you could add a full stop to the end of the statements?

cheoksuanne commented on src/main/java/bot/Parser.java:
```diff
@@ -0,0 +1,176 @@
+                indexDateDelimiter = i;
+            }
+        }
+        try{
```
I think there should be a space between try and {, and in lines 139 and 143 for the for loop

cheoksuanne commented on src/main/java/bot/command/AddCommand.java:
```diff
@@ -0,0 +1,75 @@
+            throw new IOException("Sorry, do what? Please give me a valid input." +
+                    " Thank you.");
+        }
+
```
i think you can remove this line

cheoksuanne commented on src/main/java/bot/task/Event.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    /**
+     * Serialises the object.
```
i think there should be a line between the description and the return value

cheoksuanne reviewed (commented):
I like your code, I think its very neat! Good job :)

Trav1sT committed:
Fix coding style on peer review

Trav1sT committed:
Add javafx for gradle and package files into bot

Trav1sT committed:
Configure gradle to launch UIWindow instead of CLI

Trav1sT committed:
Change file structure

Trav1sT committed:
Move test classes to a new test source

Trav1sT committed:
Fix dialogbox gets truncated issue

Trav1sT committed:
Fix bug where user aint greeted in UI window

Trav1sT committed:
Configure for jar file

Trav1sT committed:
Merge branch 'configure-jar' into master

Trav1sT committed:
Fix expected output of test case

Trav1sT committed:
Add assertions

Trav1sT committed:
Fix syntax based on java checkstyle plugin

Trav1sT committed:
Merge pull request #1 from Trav1sT/branch-A-Assertions

Add assertions

Trav1sT committed:
Merge pull request #2 from Trav1sT/branch-A-CodeQuality

Fix syntax based on java checkstyle plugin

Trav1sT committed:
Add function to allow different user input format
User can add regex expression and its corresponding expression in datetime format at src/ main/resources/dateFormat/ so that the bot can parse their preferred date format. It has to follow the syntax:
<regex expression> | <datetime format expression>
Each stmt should start at a new line

Trav1sT committed:
Fix bug where input bye does not close window

Trav1sT committed:
Add 2 additional dateformats

Trav1sT committed:
Refactor natural date parser

Trav1sT committed:
Add exception handling and parse serialised user data

Trav1sT committed:
Add test cases

Trav1sT committed:
Add side pane that displayed title

Trav1sT committed:
Add screenshot

Trav1sT committed:
ADd user guide

Trav1sT committed:
Fix typo

Trav1sT committed:
Change dateformat.txt to be read as inputStream for Jar

Trav1sT committed:
Fix typo

Trav1sT committed:
Fix typo

Trav1sT committed:
Add this in because for some reason the sys never register

Trav1sT committed:
Add img in

Trav1sT committed:
Reuploading

Trav1sT committed:
Add Ui again

Trav1sT committed:
Update for mac compatibility

Trav1sT committed:
Add examples

Trav1sT committed:
Pass java checkstyle

Trav1sT committed:
Fix import stmt

Trav1sT committed:
Add types of task

Trav1sT committed:
Fix typoes
